compiler/daml-lf-verify/src/DA/Daml/LF/Verify/Read.hs

```diff
@@ -0,0 +1,77 @@
+
+-- Copyright (c) 2020 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+-- SPDX-License-Identifier: Apache-2.0
+
+-- TODO: There is a lot of copying going on here from TsCodeGenMain.hs.
+-- A nicer alternative would be to just change the exports from this module.
+
+-- | Reading dar files for DAML LF verification.
+module DA.Daml.LF.Verify.Read
+  ( readPackages
+  , optionsParserInfo
+  , Options(..)
+  ) where
+
+import qualified DA.Daml.LF.Proto3.Archive as Archive
+import qualified DA.Daml.LF.Reader as DAR
+import qualified Data.ByteString as B
+import qualified Data.ByteString.Lazy as BSL
+import qualified Data.Text.Extended as T
+import qualified "zip-archive" Codec.Archive.Zip as Zip
+import Control.Monad.Extra
+import Options.Applicative
+
+import DA.Daml.LF.Ast
+
+data Options = Options
+    { optInputDar :: FilePath
+    , optChoiceTmpl :: String
+    , optChoiceName :: String
+    , optFieldTmpl :: String
+    , optFieldName :: String
+    }
+
+optionsParser :: Parser Options
+optionsParser = Options
```

Comment:
Having so many positional arguments will probably become an issue in the future because it's easy to mix them up. But let's fix that in a follow-up PR.